model-optimizer/extensions/front/output_cut.py

```diff
@@ -32,3 +32,16 @@ def run_before(self):
 
     def find_and_replace_pattern(self, graph: Graph):
         add_output_ops(graph, graph.graph['packed_outputs'], inputs=graph.graph['user_shapes'])
+
+        for node in graph.get_op_nodes():
+            if node.soft_get('needs_removal') is True:
+                in_node = None
+                if node.in_nodes():
+                    in_node = node.in_node(0)
+                fw_info = None
+                if node.in_edges() and 'fw_tensor_debug_info' in node.in_edge(0):
+                    fw_info = node.in_edge(0)['fw_tensor_debug_info']
+                graph.erase_node(node)
+
+                if fw_info is not None and in_node is not None and in_node.out_edges():
+                    in_node.out_edge(0)['fw_tensor_debug_info'] = fw_info
```

Comment:
The `in_node` may be connected to `node` using non-zero output port. So we should get the port number from input edge (it is stored in the "out" attribute of the edge) and use it instead of hardcoded value 0. Also, in line 46 the check `and in_node.out_edges()` is not applicable.